### PR TITLE
feat: documentation website

### LINKED ISSUE #366: [cdk8s] Add documentation to cdk8s website
### Description of the feature or enhancement:

We need an official documentation story on the cdk8s website with proper navigation and path.

### Other:

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by eladb**
Renders a documentation website for CDK8s from the contents of `docs/` using [mkdocs-material].

[mkdocs-material]: https://squidfunk.github.io/mkdocs-material

I've moved most of the content from README files across the repo to `docs/` and organized to a few categories. I've also converted some of the docs to use mkdocs features like tabs (e.g. we now have a single __getting started__ topic).

See [live sample](http://eladb.github.com/cdk8s-docsite-example/docs).

Resolves #366 

---

### TOC

```
Overview
Getting Started
Concepts
  ApiObject
  Chart
  Constructs
  Dependencies
  Helm Support
  Include
  Testing
CDK8s+
  Overview
  ConfigMap
  Container
  Deployment
  Ingress
  Job
  Pod
  Secret
  ServiceAccount
  Service
  Volume
API Reference
  cdk8s
  cdk8s-plus
  cdk8s-cli
Examples
  Hello, Kubernetes!
  WebService Abstraction
  CRDs
  Podinfo
  Elasticsearch (CDK8s+)
  Ingress (CDK8s+)
Project
  Support
  Roadmap
  Changelog
  Community
  Contributing
```

### TODO

- [x] Update cdk8s.io (and any other old references) to point to the new docsite
- [x] Fix all mkdocs warnings.
- [x] Remove tabs
- [x] Copy API.md to "references", CONTRIBUTING.md, etc
- [x] Versions
- [x] Logo

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by iliapolo** [comment]
This is kind of funky - couldn't actually find references to the CAPS in the code. Do you know whats happening?

Screen Shot 2020-11-10 at 4 56 52 PM

File: docs/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,910 @@
...
+
+    ```console
+    $ npm run watch
+    ```
+
+=== "Python"
+    ```console
+    $ mkdir hello
+    $ cd hello
+    $ CDK8s init python-app
```
Comment (iliapolo) [inline docs/getting-started.md:69]: Found it.

```suggestion
    $ cdk8s init python-app
```

**Comment by eladb** [comment]
> Do you know whats happening?

It's called "search and replace"... Let me check

File: docs/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,910 @@
...
+    creating a new project from template: python-app
+    ...
+    ```
+
+=== "Java"
+    ```console
+    $ mkdir hello
+    $ cd hello
+    $ CDK8s init java-app
```
Comment (iliapolo) [inline docs/getting-started.md:78]: Same

File: docs/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,910 @@
...
+
+## New Project
+
+Now, we'll use the `cdk8s init` command to create a new CDK8s app:
+
+=== "TypeScript"
+    ```console
+    $ mkdir hello
+    $ cd hello
+    $ CDK8s init typescript-app
```
Comment (iliapolo) [inline docs/getting-started.md:52]:
```suggestion
    $ cdk8s init typescript-app
```

**Comment by eladb** [comment]
@iliapolo fixed - spare me the manual deployment to the staging site :-)

**Review by iliapolo — APPROVED** [review]
👍

**Pull request merged**